src/test/groovy/org/powerflows/dmn/engine/configuration/DefaultDecisionEngineConfigurationGroovySpec.groovy

```diff
@@ -0,0 +1,188 @@
+
+    void 'should evaluate empty collection rules result using default decision engine'() {
+        given:
+        final DecisionEngine decisionEngine = decisionEngineConfiguration.configure()
+        final String decisionFileName = 'decision-with-groovy-expressions-only.yml'
+        final InputStream decisionInputStream = this.class.getResourceAsStream(decisionFileName)
+        final Decision decision = decisionReader.read(decisionInputStream)
+
+        final Map<String, Object> variables = [:]
+        variables.put('x', 2)
+        variables.put('y', 3)
+        variables.put('p', 0)
+        variables.put('q', 0)
+        final DecisionContextVariables decisionContextVariables = new DecisionContextVariables(variables)
+
+        when:
+        final DecisionResult decisionResult = decisionEngine.evaluate(decision, decisionContextVariables)
+
+        then:
+        decisionResult != null
+
+        with(decisionResult) {
+            !isSingleEntryResult()
+            !isSingleRuleResult()
+            isCollectionRulesResult()
+            getCollectionRulesResult().isEmpty()
+        }
+    }
+
+    void 'should evaluate single rule and single entry result using default decision engine'() {
+        given:
+        final DecisionEngine decisionEngine = decisionEngineConfiguration.configure()
+        final String decisionFileName = 'decision-with-groovy-expressions-only.yml'
+        final InputStream decisionInputStream = this.class.getResourceAsStream(decisionFileName)
+        final Decision decision = decisionReader.read(decisionInputStream)
+
+        final Map<String, Object> variables = [:]
+        variables.put('x', 1)
+        variables.put('y', 1)
+        variables.put('p', 0)
+        variables.put('q', 0)
+        final DecisionContextVariables decisionContextVariables = new DecisionContextVariables(variables)
+
+        when:
+        final DecisionResult decisionResult = decisionEngine.evaluate(decision, decisionContextVariables)
+
+        then:
+        decisionResult != null
+
+        with(decisionResult) {
+            isSingleEntryResult()
+            isSingleRuleResult()
+            !isCollectionRulesResult()
+            getSingleRuleResult().getEntryResults().size() == 1
+        }
+
+        final EntryResult entryResult = decisionResult.getSingleRuleResult().getEntryResults()[0]
+        with(entryResult) {
+            getName() == 'outputOne'
+            !getValue()
+        }
+    }
+
+    void 'should evaluate single rule result using default decision engine'() {
+        given:
+        final DecisionEngine decisionEngine = decisionEngineConfiguration.configure()
+        final String decisionFileName = 'decision-with-groovy-expressions-only.yml'
+        final InputStream decisionInputStream = this.class.getResourceAsStream(decisionFileName)
+        final Decision decision = decisionReader.read(decisionInputStream)
```

Comment:
this is repeated in each test make it in setup() or even setupSpec() as the decision doesn't need to be re read...

Reply:
True, however I want to change an approach to test particular evaluator in a separated issue. See my comment to GroovyExpressionEvaluationProvider.java

Reply:
the comment is vague and can't see it's relationship to proper initialization of tests here...

Reply:
Check it now please.